internal/pkg/machine/controllers/kubernetes_node.go

```diff
@@ -136,15 +137,15 @@ func (ctrl *KubernetesNodeController) Run(ctx context.Context, r controller.Runt
 		case <-r.EventCh():
 		}
 
-		config, err := safe.ReaderGetByID[*config.MachineConfig](ctx, r, config.V1Alpha1ID)
-		if err != nil {
-			if state.IsNotFoundError(err) {
-				continue
-			}
-
+		config, err := machineconfig.GetComplete(ctx, r)
+		if err != nil && !state.IsNotFoundError(err) {
 			return err
 		}
 
+		if config == nil || config.Provider().Machine() == nil {
```

Comment:
I guess this `config.Provider().Machine() == nil` check isn't necessary here anymore.